packages/bpk-component-popover/src/BpkPopover.tsx

```diff
@@ -0,0 +1,240 @@
+  closeButtonIcon?: boolean;
+  closeButtonProps?: Object;
+  isOpen?: boolean;
+  labelAsTitle?: boolean;
+  padded?: boolean;
+  placement?: 'top' | 'right' | 'bottom' | 'left';
+  showArrow?: Boolean;
+  target: ReactElement<any>;
+};
+
+const BpkPopover = ({
+  children,
+  className = null,
+  closeButtonIcon = true,
+  closeButtonProps = {},
+  closeButtonText,
+  id,
+  isOpen = false,
+  label,
+  labelAsTitle = false,
+  onClose,
+  padded = true,
+  placement = 'bottom',
+  showArrow = true,
+  target,
+  ...rest
+}: Props) => {
+  const [isOpenState, setIsOpenState] = useState(isOpen);
+  const arrowRef = useRef(null);
+
+  const { context, floatingStyles, refs } = useFloating({
+    open: isOpenState,
+    onOpenChange: setIsOpenState,
+    placement,
+    middleware: [
+      showArrow && offset(17),
+      flip({ crossAxis: true }),
+      shift(),
+      showArrow && arrow({ element: arrowRef }),
+    ],
+    whileElementsMounted: autoUpdate,
+  });
+
+  const click = useClick(context);
+  const dismiss = useDismiss(context);
+
+  // Merge all the interactions into prop getters
+  const { getFloatingProps, getReferenceProps } = useInteractions([
+    click,
+    dismiss,
+  ]);
+
+  const targetElement = isValidElement(target) ? (
+    cloneElement(target, {
+      ...getReferenceProps(),
+      // @ts-ignore - we're adding a popover ref to the target element so we can position the popover relative to it
+      ref: refs.setReference,
+    })
+  ) : (
+    <div ref={refs.setReference} {...getReferenceProps()}>
+      {target}
+    </div>
+  );
+  const classNames = getClassName('bpk-popover', className);
+  const bodyClassNames = getClassName(padded && 'bpk-popover__body--padded');
+
+  const labelId = `bpk-popover-label-${id}`;
+
+  return (
+    <>
+      {targetElement}
+      {isOpenState && (
+        <FloatingFocusManager context={context} modal={false}>
+          <div
+            ref={refs.setFloating}
+            style={floatingStyles}
+            {...getFloatingProps()}
+          >
+            <TransitionInitialMount
+              appearClassName={getClassName('bpk-popover--appear')}
+              appearActiveClassName={getClassName('bpk-popover--appear-active')}
+              transitionTimeout={200}
+            >
+              <section
+                id={id}
+                tabIndex={-1}
+                role="dialog"
+                aria-labelledby={labelId}
+                className={classNames}
+                {...rest}
+              >
+                {showArrow && (
+                  <FloatingArrow
+                    ref={arrowRef}
+                    context={context}
+                    id={ARROW_ID}
+                    // eslint-disable-next-line @skyscanner/rules/forbid-component-props
+                    className={getClassName('bpk-popover__arrow')}
+                    role="presentation"
+                    stroke={surfaceHighlightDay}
+                    strokeWidth={0.0625}
```

Comment:
Hmm could be use a token here or declare it as a constant at the top?

Reply:
I can move it to a const at the top. We can't use a token here sadly as Backpack Foundations only provide them with a string and include the word `rem` in them whereas floating UI expects a number